File: code/python/539.py
# https://leetcode.com/problems/minimum-time-difference/
class Solution:
    def findMinDifference(self, timePoints: List[str]) -> int:
        time_set = set()
        time_int = []
        for point in timePoints:
            tmp = int(point[0:2]) * 60 + int(point[3:5])
            if tmp in time_set:
                return 0
            time_set.add(tmp)
            time_int.append(tmp)
        time_int.sort()
        res = 24 * 60 - time_int[-1] + time_int[0]
        for i in range(1, len(time_int)):
            res = min(res, time_int[i] - time_int[i - 1])
        return res

File: code/cpp/539.cpp
// https://leetcode.com/problems/minimum-time-difference/
class Solution {
public:
    int findMinDifference(vector<string>& timePoints) {
        size_t i = 0, n = timePoints.size(), tmp;
        int res;
        vector<int> timeInt(n);
        for (const string& s : timePoints) {
            tmp = ((s[0] - '0') * 10 + s[1] - '0') * 60 + (s[3] - '0') * 10 + (s[4] - '0');
            timeInt[i] = tmp;
            i++;
        }
        sort(timeInt.begin(), timeInt.end());
        res = 24 * 60 - timeInt[n - 1] + timeInt[0];
        for (int i = 1; i < n; i++)
            res = min(res, timeInt[i] - timeInt[i - 1]);
        return res;
    }
};
